### DISCUSSION TITLE
Trying to Mount cdrom (Sr0)

### DISCUSSION THREAD

**Discussion by mxrss2** [siderolabs/talos#9897] [Q&A]
I am trying to mount a (/dev/sr0) device into my worker nodes.

Background

Each worker node would have a cdrom and these privliged containers would eject and read -- i can see the mount in /var/mnt/cdrom but when i 

`talosctl list /var/mnt/cdrom` i cant see any contents on the cdrom. 

The envrionment is in a VM (Vmware fusion)

**Comment by smira** [comment]
It's not quite clear what the question is.


As the mount is performed by a container, by default that mount will not propagate back to the host. If you want the mount to propagate, you need to mount the host /var into the container with mount propagation, mount the CD-ROM in a container under /var, and then it would propagate back to the host

**Reply by mxrss2** [reply]
@smira  thank you for getting back to me. 

The problem i have is that going to the location on disk `/var/mnt/cdrom` doesnt show any files. I just see this. `.` i had a config similar to this


```
yamlencode({
            machine = {
                install = {
                    disk = "/dev/nvme0n1"
                    wipe = true
                }
                disks = [{
                    device = "/dev/sr0"
                    
                }]
            }
        })

```

**Reply by mxrss2** [reply]
what i am trying to do is to get talos to see the cdrom drive so that i can connect with a privileged container to eject the cdrom and to read its contents.

**Reply by smira** [reply]
The `.machine.user.disks` can't be used for CDROMs.

Talos sees your CDROM drive, and there's no problem on Talos side.

A privileged container should be able to see your CD-ROM, but you can also do https://www.talos.dev/v1.8/kubernetes-guides/configuration/device-plugins/. This is not specific to Talos in any way.
